### PR TITLE
change sku tier to not force new

### PR DESCRIPTION
**Pull request by diklatze**
We have added a feature for single-click migration that upgrade the SKU and there is no need to force new.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @diklatze  - could we update the "update" test to ensure its changing the sku to verify this works (and continues to do so)

**Comment by diklatze** [comment]
> Thanks @diklatze - could we update the "update" test to ensure its changing the sku to verify this works (and continues to do so)

I have added a test. Is that what you mean?

File: internal/services/firewall/firewall_resource.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func resourceFirewall() *pluginsdk.Resource {
 			"sku_tier": {
 				Type:     pluginsdk.TypeString,
 				Required: true,
-				ForceNew: true,
+				ForceNew: false,
```
Comment (katbyte) [inline internal/services/firewall/firewall_resource.go:80]: we can just remove this line
```suggestion
```

File: internal/services/firewall/firewall_resource_test.go [hunk 2]
```diff
@@ -194,6 +195,31 @@ func TestAccFirewall_withZones(t *testing.T) {
 	})
 }
 
+func TestAccFirewall_skuTier(t *testing.T) {
```
Comment (katbyte) [inline internal/services/firewall/firewall_resource_test.go:198]: we should be more explicit here
```suggestion
func TestAccFirewall_skuTierUpdate(t *testing.T) {
```

File: internal/services/firewall/firewall_resource_test.go [hunk 3]
```diff
@@ -194,6 +195,31 @@ func TestAccFirewall_withZones(t *testing.T) {
...
+	data := acceptance.BuildTestData(t, "azurerm_firewall", "test")
+	r := FirewallResource{}
+	skuTier := string(network.AzureFirewallSkuTierStandard)
+	skuTierUpdate := string(network.AzureFirewallSkuTierPremium)
```
Comment (katbyte) [inline internal/services/firewall/firewall_resource_test.go:202]: can we just use constants here instead of pulling the value from the sdk?

**Review by katbyte — CHANGES_REQUESTED** [review]
@diklatze - running tests for this and i am getting a failure:

image

you'll also need to update the docs, once thats all fixed up we can get this merged

File: internal/services/firewall/firewall_resource_test.go (hunk 3 above)
Comment (diklatze) [reply internal/services/firewall/firewall_resource_test.go]: @katbyte I pushed all the changes that you mentioned. Thanks!

**Review by stephybun — APPROVED** [review]
Thanks for this @diklatze, the test was still failing with the same error as before, I hope you don't mind I went ahead and pushed a fix for it so that we can get this in to the next release.
On that note, LGTM 👍

**Pull request merged**